src/main/java/org/eclipse/xpanse/terraform/boot/terraform/TerraformInstaller.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ */
+
+package org.eclipse.xpanse.terraform.boot.terraform;
+
+import jakarta.annotation.Resource;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URL;
+import java.nio.channels.Channels;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.xpanse.terraform.boot.models.exceptions.InvalidTerraformToolException;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.retry.annotation.Backoff;
+import org.springframework.retry.annotation.Retryable;
+import org.springframework.stereotype.Component;
+
+/**
+ * Bean help to install terraform with specific version.
+ */
+@Slf4j
+@Component
+public class TerraformInstaller {
+
+    @Value("${terraform.download.base.url:https://releases.hashicorp.com/terraform}")
+    private String terraformDownloadBaseUrl;
+    @Value("${terraform.install.dir:/opt/terraform}")
+    private String terraformInstallDir;
+    @Resource
+    private TerraformToolHelper terraformToolHelper;
+
+    /**
+     * Get the executable binary path of the terraform tool with version in the environment.
+     * Install the executable binary with specific version into the path in the environment when
+     * it is not installed.
+     */
+    @Retryable(retryFor = InvalidTerraformToolException.class,
+            maxAttemptsExpression = "${spring.retry.max-attempts}",
+            backoff = @Backoff(delayExpression = "${spring.retry.delay-millions}"))
+    public String getTerraformExecutorPathByVersion(String version) {
+        if (StringUtils.isBlank(version)) {
+            return "terraform";
+        }
+        String versionNumber = terraformToolHelper.getTerraformVersionNumber(version);
+        String terraformExecutorName =
+                terraformToolHelper.getTerraformExecutorName(versionNumber);
+        File terraformExecutorFile =
+                new File(this.terraformInstallDir + File.separator + terraformExecutorName);
+        if (terraformToolHelper.checkIfTerraformInstalled(terraformExecutorFile, versionNumber)) {
+            log.info("Terraform with version {} is already installed in the dir {}.",
```

Comment:
if a version lesser or greater condition is matched, then we must log that specific version that will be used for deploying. 

I just realized that, probably we must also return the exact version used to deploy also in the async response back. It will be useful information in case of issues.